uhura/caches.py

```diff
@@ -26,7 +26,7 @@ def update(self, obj):
         raise NotImplementedError()
 
 
-class Cacheable(Protocol):
```

Comment:
I'd prefer to make this an ABC and then explicitly subclass in Readable and Writable. This will fail otherwise if we add mypy.